File: test/test_generator.py
from hydra import *

import unittest

class TestGenerator(unittest.TestCase):

    def test_generator(self):
        gDescriptor("D0 => K_S0 K+ K-")
        gen = Generator("Generator")

        ev = Event()
        gen( ev )

        self.assertIsNot( ev.mother().time() , 0 )
        self.assertEqual( ev.particles().size() , 4 )

File: test/test_generator.cpp
#include "catch.hpp"

#include "generator.h"
#include "descriptor.h"
#include "event.h"

TEST_CASE( "Generator will generate events", "[Generator]") {
  // Set Decay.
  gDescriptor("D0 => K_S0 K+ K-");
  // Create Generator.
  Generator gen("Generator");

  SECTION("Generate decay") {
    // Define event.
    Event ev;
    // Generate event.
    gen(ev);
    REQUIRE( ev.mother().time() > 0 );
    REQUIRE( ev.particles().size() == 4 );
  }
}